packages/test/src/browser/view/test-run-widget.tsx

```diff
@@ -216,6 +218,12 @@ export class TestRunTreeWidget extends TreeWidget {
                     onDidAddTestOutput: Event.map(node.parent.run.onDidChangeTestOutput, evt => evt.filter(item => item[0] === node.item).map(item => item[1]))
                 };
                 this.uiModel.selectedTestState = node.parent.run.getTestState(node.item);
+                if (this.uiModel.selectedTestState && TestFailure.is(this.uiModel.selectedTestState.state)) {
```

Comment:
Shouldn't this code live in the `uiModel` as it's the owner of the `selectedTestState`?

Reply:
Thanks for noticing! Indeed, that should happen in the uiModel. I will update the code and move this part of the code on the selectedTestState mutator.